pkg/cloudprovider/providers/vsphere/vclib/utils.go

```diff
@@ -0,0 +1,148 @@
+package vclib
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func getFinder(dc *Datacenter) *find.Finder {
+	finder := find.NewFinder(dc.Client(), true)
+	finder.SetDatacenter(dc.Datacenter)
+	return finder
+}
+
+func formatVirtualDiskUUID(uuid string) string {
+	uuidwithNoSpace := strings.Replace(uuid, " ", "", -1)
+	uuidWithNoHypens := strings.Replace(uuidwithNoSpace, "-", "", -1)
+	return strings.ToLower(uuidWithNoHypens)
+}
+
+func createDiskSpec(ctx context.Context, diskPath string, vm *VirtualMachine, dsObj *Datastore, volumeOptions VolumeOptions) (*types.VirtualDisk, types.BaseVirtualDevice, error) {
```

Comment:
Makes sense. i will move it to VM obj.

Also, getSCSIControllersOfType is required for you to decide whether a particular SCSI controller is already present on the VM and it is available for us to use it. If not available we need to create one to attach the disk with that controller.